README.md

```diff
@@ -38,6 +38,14 @@ manager behavior.
 - Persists newly registered data in-memory (the original library JSON is
   not updated).
 
+- Simulate network delay with the `simulated_query_latency_ms` setting.
+  > **Note**
+  >
+  > Simulated query latency defaults to 10ms. It uses `time.sleep`
+  > behind the scenes, which may have GIL implications on your platform,
+  > refer to the [official
+  > docs.](https://docs.python.org/3/library/time.html#time.sleep)
```

Comment:
No your not, I misremembered another post I was referencing and must have substituted it for the offical docs in my head.

I wonder if we should mention the GIL at all here. There's a similar discussion on this point about `sleep` in the official python docs which was very interesting to read : https://bugs.python.org/issue23251

I'd be concerned about making GIL assertions based on implicit behaviour of `sleep` when that behaviour isn't documented in either the interface nor the function documentation. Maybe we do need to test it more thoroughly if we're gonna be talking about GIL stuff, although Antoines comment from that thread comes to mind : 

> Please let's stop it. Mentioning the GIL in every place in the documentation will only make people more worried and confused about something they shouldn't be confused or worried about.

Reply:
Happy to update the ACs to not talk about the GIL and we just document we use `sleep` so people can go hunt on their own - if that simplifies everything?

Reply:
Yeah I'm glad we're on the same page, that's what I was thinking, defer the GIL and threading implications to `sleep`, since we're entirely deriving our behaviour from that. (Albeit it would be nicer if the python folks mentioned their GIL implications in their method documentations, implicit assumptions drive me up the wall.)

Reply:
732bdf4fc616c6067d99cadd9038ecb14de76bb9